### PR TITLE
flake: provide containerized builds

### PR DESCRIPTION
**Pull request by NotAShelf**
Provides a simple docker build for the minimal `nix` configuration. More builds may be provided in the future.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        uses: DeterminateSystems/nix-installer-action@main
+      - name: Magic Nix Cache
+        uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: Build Docker Image
+        run: nix build .#docker-nix --print-build-logs
+
+      - name: Load & Tag Built Docker Image
+        run: |
+          docker load -i result ; docker images &&
```
Comment (Vagahbond) [inline .github/workflows/docker.yml:37]: `docker images` is there for debugging purpose, right ? maybe remove it /

File: flake/packages.nix [hunk 2]
```diff
@@ -19,16 +18,43 @@
...
           ${pkgs.xdg_utils}/bin/xdg-open ${docs.manual.html}
         '';
 
-        # nvim configs
+        # Exposed neovim configurations
         nix = config.legacyPackages.neovim-nix;
         maximal = config.legacyPackages.neovim-maximal;
         default = config.legacyPackages.neovim-nix;
+
+        # Publisged docker images
```
Comment (Vagahbond) [inline flake/packages.nix:33]: Publisged

File: .github/workflows/docker.yml (hunk 1 above)
Comment (NotAShelf) [reply .github/workflows/docker.yml:37]: oh yeah, goodo ne

File: flake/packages.nix (hunk 2 above)
Comment (NotAShelf) [reply flake/packages.nix:33]: idk what's the problem tbh

**Review by FrothyMarrow — APPROVED** [review]

**Review by horriblename — APPROVED** [review]

**Pull request merged**